From e2ed1fc059325c3df993ab6c3b5952b87ef54c40 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 13 Jun 2019 18:42:40 +0200
Subject: [PATCH 1/1] efi_loader: GetTime() must return EFI_UNSUPPORTED

If the GetTime() runtime service is not supported, EFI_UNSUPPORTED has to
be returned.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_runtime.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/efi_loader/efi_runtime.c b/lib/efi_loader/efi_runtime.c
index 9ede42a728..bf730d0ff6 100644
--- a/lib/efi_loader/efi_runtime.c
+++ b/lib/efi_loader/efi_runtime.c
@@ -366,8 +366,7 @@ efi_status_t __weak __efi_runtime EFIAPI efi_get_time(
 			struct efi_time *time,
 			struct efi_time_cap *capabilities)
 {
-	/* Nothing we can do */
-	return EFI_DEVICE_ERROR;
+	return EFI_UNSUPPORTED;
 }
 
 /**
-- 
2.20.1

